### PR TITLE
Client Count Docs Updates/Cleanup

### PR DESCRIPTION
**Pull request by divyaac**
### Description
Client count documentation updates. 

The main objectives:
- Remove any version related descriptions; the docs team wants to train customers to use the versioned docs for this information.
- Try to condense multiple bullets into as few as possible.
- Remove repetitive information, and remedy any incorrect information.

### TODO only if you're a HashiCorp employee
- [ ] **Labels:** If this PR is the CE portion of an ENT change, and that ENT change is
  getting backported to N-2, use the new style `backport/ent/x.x.x+ent` labels
  instead of the old style `backport/x.x.x` labels.
- [ ] **Labels:** If this PR is a CE only change, it can only be backported to N, so use
  the normal `backport/x.x.x` label (there should be only 1).
- [ ] **ENT Breakage:** If this PR either 1) removes a public function OR 2) changes the signature
  of a public function, even if that change is in a CE file, _double check_ that
  applying the patch for this PR to the ENT repo and running tests doesn't
  break any tests. Sometimes ENT only tests rely on public functions in CE
  files.
- [ ] **Jira:** If this change has an associated Jira, it's referenced either
  in the PR description, commit message, or branch name.
- [ ] **RFC:** If this change has an associated RFC, please link it in the description.
- [ ] **ENT PR:** If this change has an associated ENT PR, please link it in the
  description. Also, make sure the changelog is in this PR, _not_ in your ENT PR.

### CODE REVIEW

File: website/content/api-docs/system/internal-counters.mdx
Comment (divyaac) [inline website/content/api-docs/system/internal-counters.mdx:100]: Deleted this start_time/end_time information because it duplicates info that's there when we explain the parameters at the bottom with the API.

File: website/content/api-docs/system/internal-counters.mdx
Comment (divyaac) [inline website/content/api-docs/system/internal-counters.mdx:226]: This addresses the confusion in this [thread](https://hashicorp.slack.com/archives/C0282NFTX4P/p1721231854554999)

File: website/content/api-docs/system/internal-counters.mdx
Comment (miagilepner) [inline website/content/api-docs/system/internal-counters.mdx:234]:
```suggestion
current month. The last element in the `months` response stanza will signify the current month. Furthermore, the
```

**Review by miagilepner — APPROVED** [review]
LGTM, but we should also get @schavis's approval before merging

**Review by yhyakuna — APPROVED** [review]

**Pull request merged**
